src/wrapper/map.rs

```diff
@@ -35,3 +35,41 @@ pub fn map_put(env: NIF_ENV, map: NIF_TERM, key: NIF_TERM, value: NIF_TERM) -> O
     }
     Some(result)
 }
+
+pub struct MapIterator {
```

Comment:
[The idea](https://github.com/hansihe/Rustler/blob/master/src/wrapper/mod.rs#L3-L6) behind the wrapper module was to have functions be as close as possible to the original C api, while being reasonably safe to use. I am therefore a bit reluctant to put a iterator implementation in here. What do you think?